code/_onclick/xeno.dm

```diff
@@ -45,10 +45,14 @@
 				SPAN_DANGER("You swipe at [target]!"), null, 5, CHAT_TYPE_XENO_COMBAT)
 	return TRUE
 
-/mob/living/carbon/Xenomorph/RangedAttack(var/atom/A)
+/mob/living/carbon/Xenomorph/RangedAttack(var/atom/A, var/list/mods)
 	. = ..()
 	if (.)
 		return
+	if(client?.prefs && HAS_FLAG(client.prefs.toggle_prefs, TOGGLE_XENO_RANGED_ATTACK_CLICK))
+		var/caste_result = caste.handle_ranged_attack(src, A, mods)
+		if(caste_result != null)
```

Comment:
```suggestion
		if(caste_result)
```

Reply:
nah, checking explicitly for null so people can return FALSE if they want to for whatever reason